core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -98,6 +98,11 @@ public <T> T getService(Class<T> type, boolean isOptional) {
 
     @Override
     public <T> void registerService(Class<T> type, T service) {
+        registerServiceRaw(type, service);
```

Comment:
typically I think people should use the typed one to increase type safety and avoid runtime crashes.
I had to introduce the `*raw` one because `Method#getReturnType()` returns an unbound wildcard `Class<?>`. 
I'll just cast that to a `T`, then the `*raw` method can be removed.